app/leave-app/backend/modules/authorization/authorization.bal

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/http;
+import ballerina/jwt;
+import ballerina/log;
+
+// Allowed user roles of the API
+configurable string[] userRoles = ?;
+// Allowed admin roles of the API
+public configurable string[] adminRoles = ?;
+
+# To handle authorization for each resource function invocation.
+public isolated service class JwtInterceptor {
+
+    *http:RequestInterceptor;
+
+    isolated resource function default [string... path](http:RequestContext ctx, http:Request req)
+        returns http:NextService|http:Unauthorized|http:BadRequest|http:Forbidden|http:InternalServerError|error? {
+
+        final string method = req.method;
+        if method == http:OPTIONS {
+            return ctx.next();
+        }
```

Comment:
When an application frontend communicates with this service, it will send a `x-jwt-assertion` header in its context and will be able to consume this service endpoints.
Is it possible to do service to service communication here, given that the invoker does not have a `x-jwt-assertion` header in its context?

Reply:
It only applies for `http:OPTIONS` method. Other HTTP methods (like `GET` or `POST`) won’t be affected by this logic.

Reply:
Resolved as we discussed in the online meeting.